basil/utils/USBBinds.py

```diff
@@ -228,6 +235,8 @@ def modify_basil_config(conf, log, skip_binds=[], save_modified=None):
             port = None
 
         instruments.append({
+            "name": tf["name"],
+            "type": "", # Standard instruments with *IQN? command
```

Comment:
What is happening here exactly? Is this a way to define different devices with non-`*IDN?` identifier commands?

Reply:
This is done to identify serial transfer layers which uses implemented non-`*IDN?` devices. Without the addition of "name" and "type" in this temporary instrument dictionary the linking in the code block below would not work and thus also not the logic which you commented in your third comment.

**TL;DR:** The serial transfer layer does not know if it is used for e.g. an SHQ 122M. Therefore this part is needed to identify an instrument correctly.